packages/hardhat-core/test/internal/hardhat-network/provider/utils/runFullBlock.ts

```diff
@@ -69,57 +67,27 @@ export async function runFullBlock(
     }
   );
 
-  // TODO uncomment and fix this
-  // forkedNode["_vmTracer"].disableTracing();
-
-  const afterBlockEvent = await runBlockAndGetAfterBlockEvent(
-    // TODO remove "as any" and make this work with VMAdapter
-    forkedNode["_vm"] as any,
-    {
-      block,
-      generate: true,
-      skipBlockValidation: true,
-    }
-  );
-
-  const modifiedBlock = afterBlockEvent.block;
-
-  // TODO remove "as any" and make this work with VMAdapter
-  await (forkedNode["_vm"] as any).blockchain.putBlock(modifiedBlock);
-  await (forkedNode["_vm"] as any).putBlock(modifiedBlock);
-  await forkedNode["_saveBlockAsSuccessfullyRun"](
-    modifiedBlock,
-    afterBlockEvent as any // TODO remove this as any
-  );
-
-  const newBlock = await forkedNode.getBlockByNumber(blockToRun);
-
-  if (newBlock === undefined) {
-    assert.fail();
-  }
-
-  await assertEqualBlocks(newBlock, afterBlockEvent, rpcBlock, forkClient);
-}
-
-async function runBlockAndGetAfterBlockEvent(
-  vm: VM,
-  runBlockOpts: RunBlockOpts
-): Promise<AfterBlockEvent> {
-  let results: AfterBlockEvent;
-
-  function handler(event: AfterBlockEvent) {
-    results = event;
+  const vm = forkedNode["_vm"];
```

Comment:
@fvictorio could you please have a look at this too?

Reply:
I was unfamiliar with the previous test and since Gene changed it, I wanted you to validate the TS